### PR TITLE
Added comprehensive tests for scala collections, three of which currently fail

### PR DESCRIPTION
**Pull request by jdanbrown**

### CODE REVIEW

File: src/test/scala/com/codahale/jerkson/tests/JsonGenerationSpec.scala [hunk 1]
```diff
@@ -234,3 +235,146 @@ case class CaseClassWithOverloadedField(id: Long) {
...
+
+  class `An ArraySeq` {
+    def `should generate a JSON array` = {
+      generate(mutable.ArraySeq(1, 2, 3)) must beEqualTo(generate(Seq(1, 2, 3)))
+    }
+  }
+
+  class `A StringBuilder` {
+    def `should generate a JSON string` = {
+      generate(new StringBuilder("foo")) must beEqualTo(generate("foo"))
```
Comment (jdanbrown) [inline src/test/scala/com/codahale/jerkson/tests/JsonGenerationSpec.scala:263]: fails

File: src/test/scala/com/codahale/jerkson/tests/JsonGenerationSpec.scala [hunk 2]
```diff
@@ -234,3 +235,146 @@ case class CaseClassWithOverloadedField(id: Long) {
...
+
+  class `An immutable HashMap` {
+    def `should generate a JSON object` = {
+      generate(immutable.HashMap("one" -> 1)) must beEqualTo("""{"one":1}""")
+    }
+  }
+
+  class `A mutable HashMap` {
+    def `should generate a JSON object` = {
+      generate(mutable.HashMap("one" -> 1)) must beEqualTo("""{"one":1}""")
```
Comment (jdanbrown) [inline src/test/scala/com/codahale/jerkson/tests/JsonGenerationSpec.scala:365]: fails

File: src/test/scala/com/codahale/jerkson/tests/JsonGenerationSpec.scala [hunk 3]
```diff
@@ -234,3 +235,146 @@ case class CaseClassWithOverloadedField(id: Long) {
...
+    }
+  }
+
+  class `A LinkedHashMap` {
+    def `should generate a JSON object` = {
+      generate(mutable.LinkedHashMap("one" -> 1)) must beEqualTo("""{"one":1}""")
```
Comment (jdanbrown) [inline src/test/scala/com/codahale/jerkson/tests/JsonGenerationSpec.scala:371]: fails

**Comment by codahale** [comment]
Merged, kind of. I cherry-picked the bits related to the test cases, thanks.

In the future, though, I'd appreciate a pull request from a branch dedicated to the intended change.

**Comment by jdanbrown** [comment]
On Wed, Jun 1, 2011 at 21:48, codahale
reply@reply.github.com
wrote:

> In the future, though, I'd appreciate a pull request from a branch dedicated to the intended change.

Noted. I'm still figuring out git.

 Dan

**Pull request closed without merging**